projects/ks89/angular-modal-gallery/src/lib/components/previews/previews.component.ts

```diff
@@ -251,80 +235,51 @@ export class PreviewsComponent extends AccessibleComponent implements OnInit, On
-   * Private method to init both `start` and `end` to the beginning.
-   */
-  private setBeginningIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = 0;
-    this.end = Math.min(this.previewConfig.number as number, this.images.length);
-  }
-
-  /**
-   * Private method to init both `start` and `end` to the end.
-   */
-  private setEndIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images) {
-      throw new Error('Internal library error - previewConfig and images must be defined');
-    }
-    this.start = this.images.length - 1 - ((this.previewConfig.number as number) - 1);
-    this.end = this.images.length;
-  }
-
   /**
    * Private method to update both `start` and `end` based on the currentImage.
    */
-  private setIndexesPreviews(): void {
-    if (!this.previewConfig || !this.images || !this.currentImage) {
+  private setIndexesPreviews(currentImage: InternalLibImage, images: InternalLibImage[]): void {
+    if (!this.previewConfig || !images || !currentImage) {
       throw new Error('Internal library error - previewConfig, currentImage and images must be defined');
     }
-    this.start = getIndex(this.currentImage, this.images) - Math.floor((this.previewConfig.number as number) / 2);
-    this.end = getIndex(this.currentImage, this.images) + Math.floor((this.previewConfig.number as number) / 2) + 1;
+    const previewsNumber = this.previewConfig.number as number;
+    let start = getIndex(currentImage, images) - Math.floor(previewsNumber / 2);
+    // start is, at a minimum, the first index
+    if(start < 0) start = 0;
+    // end index
+    let end = start + previewsNumber;
+    // end is, at a maximum, the last index
+    if(end > images.length) {
+      start -= end - images.length;
+      if(start < 0) start = 0; // start is, at a minimum, the first index
+      end = images.length;
+    }
+    this.start = start;
+    this.end = end;
   }
 
   /**
    * Private method to update the visible previews navigating to the right (next).
    */
   private next(): void {
-    if (!this.images) {
+    if (!this.images || !this.previewConfig) {
       throw new Error('Internal library error - images must be defined');
     }
-    // check if nextImage should be blocked
-    if (this.isPreventSliding(this.images.length - 1)) {
-      return;
-    }
-
-    if (this.end === this.images.length) {
-      return;
+    if(this.end >= this.images.length) {
+      // check if nextImage should be blocked
+      const preventSliding = !!this.slideConfig && this.slideConfig.infinite === false;
+      if(preventSliding) {
+        return;
+      }
+      this.start = 0;
+    } else {
+      this.start++;
     }
-
-    this.start++;
-    this.end = Math.min(this.end + 1, this.images.length);
+    this.end = this.start + Math.min((this.previewConfig.number as number), this.images.length);
```

Comment:
Also, start and end are calculated in a co-dependant way, like in setIndexesPreviews().